extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -347,8 +347,17 @@ public function getPage() {
 			$data['ns'] = $title->getNamespace();
 
 			if ( MercuryApiMainPageHandler::shouldGetMainPageData( $isMainPage ) ) {
+
+				// TODO: remove this line after release of XW-2590 (XW-2625)
 				$data['mainPageData'] = MercuryApiMainPageHandler::getMainPageData( $this->mercuryApi );
```

Comment:
I'm not sure about removing this line. There is a lot more that is used by mercury in `mainPageData` that isn't migrated to `curatedMainPageData`. https://github.com/Wikia/mercury/blob/ee1a84ff164b62863bca8458078b5333efe4eaeb/front/main/app/templates/main-page.hbs#L11-L16
Do we have a plan to tackle that?